### PR TITLE
 [SPARK-43233] [SS] Add logging for Kafka Batch Reading for topic partition, offset range and task ID

### PR DESCRIPTION
**Pull request by siying**
### What changes were proposed in this pull request?
We add a logging when creating the batch reader with task ID, topic, partition and offset range included.
The log line looks like following:

23/04/18 22:35:38 INFO KafkaBatchReaderFactory: Creating Kafka reader partitionId=1 partition=StreamingDustTest-KafkaToKafkaTopic-4ccf8662-c3ca-4f3b-871e-1853c0e61765-source-2 fromOffset=0 untilOffset=3 queryId=b5b806c3-ebf3-432e-a9a7-d882d474c0f5 batchId=0 taskId=1

### Why are the changes needed?
Right now, for structure streaming from Kafka, it's hard to finding which task handling which topic/partition and offset range. 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Run KafkaMicroBatchV2SourceSuite and watch logging outputs contain information needed. Also does a small cluster test and observe logs.

### CODE REVIEW

**Comment by anishshri-db** [comment]
@siying - you might need to enable github actions for the tests to run

**Review by anishshri-db — APPROVED** [review]
+1 pending builds

**Comment by anishshri-db** [comment]
@HeartSaVioR - please take a look and merge after builds pass, thx !

File: connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaBatchPartitionReader.scala [hunk 1]
```diff
@@ -34,9 +36,17 @@ private[kafka010] case class KafkaBatchInputPartition(
     failOnDataLoss: Boolean,
     includeHeaders: Boolean) extends InputPartition
 
-private[kafka010] object KafkaBatchReaderFactory extends PartitionReaderFactory {
+private[kafka010] object KafkaBatchReaderFactory extends PartitionReaderFactory with Logging {
   override def createReader(partition: InputPartition): PartitionReader[InternalRow] = {
     val p = partition.asInstanceOf[KafkaBatchInputPartition]
+
+    val queryId = TaskContext.get().getLocalProperty(StreamExecution.QUERY_ID_KEY)
```
Comment (HeartSaVioR) [inline connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaBatchPartitionReader.scala:43]: nit: shall we simply add a new local variable for `TaskContext.get()`? Something like `val taskCtx = TaskContext.get()` or even just `ctx`.

File: connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaBatchPartitionReader.scala [hunk 2]
```diff
@@ -34,9 +36,17 @@ private[kafka010] case class KafkaBatchInputPartition(
...
+    val batchId = TaskContext.get().getLocalProperty(MicroBatchExecution.BATCH_ID_KEY)
+    logInfo(s"Creating Kafka reader partitionId=${TaskContext.get().partitionId()} " +
```
Comment (HeartSaVioR) [inline connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaBatchPartitionReader.scala:45]: minor: placing partitionId near to topicPartition may lead to misunderstand that partitionId refer to Kafka's one (hence redundant info.). It's Spark's partition ID. 

I'd suggest placing it after taskId, and maybe add ` for the query ` in between `untilOffset` and `queryId` to differentiate Kafka related info vs Spark/query related info.

**Review by HeartSaVioR — COMMENTED** [review]
Only minor and nit.

File: oss_spark/pom.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (HeartSaVioR) [inline oss_spark/pom.xml:1]: Maybe added by accident?

**Review by HeartSaVioR — APPROVED** [review]
+1 except unexpected pom file. Will merge once the file is removed and CI passes.

**Comment by HeartSaVioR** [comment]
Thanks! Merging to master.

**Comment by HeartSaVioR** [comment]
@siying Thanks for your first contribution to Apache Spark project. Welcome!

**Pull request closed without merging**